modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-type-date/src/main/resources/META-INF/resources/date_field.js

```diff
@@ -108,31 +108,27 @@ AUI.add(
 					render: function() {
 						var instance = this;
 
-						DateField.superclass.render.apply(instance, arguments);
-
 						var pattern = instance.get('mask');
-						pattern = pattern.replace('%d', 'DD');
-						pattern = pattern.replace('%m', 'MM');
-						pattern = pattern.replace('%Y', 'YYYY');
 
-						var moment = DDMDate.moment;
-						var IMask = DDMDate.IMask;
+						pattern = pattern.replace(/%d/, 'dd');
+						pattern = pattern.replace(/%m/, 'mm');
+						pattern = pattern.replace(/%y/, 'yy');
 
-						var element = instance.getTriggerNode().getDOM();
+						var autoCorrectedDatePipe = DDMDate.createAutoCorrectedDatePipe(pattern + 'HH:MM');
+
+						DateField.superclass.render.apply(instance, arguments);
 
-						var momentMask = new IMask(element, {
-						  mask: Date,
-						  pattern: pattern,
-						  lazy: false,
+						var element = instance.getTriggerNode().getDOM();
 
-						  groups: {
-						    YYYY: new IMask.MaskedPattern.Group.Range([1970, 2030]),
-						    MM: new IMask.MaskedPattern.Group.Range([1, 12]),
-						    DD: new IMask.MaskedPattern.Group.Range([1, 31]),
-						    HH: new IMask.MaskedPattern.Group.Range([0, 23]),
-						    mm: new IMask.MaskedPattern.Group.Range([0, 59])
-						  }
-						});
+						DDMDate.vanillaTextMask(
+							{
+								inputElement: element,
+								mask: [/\d/, /\d/, '/', /\d/, /\d/, '/', /\d/, /\d/, /\d/, /\d/],
```

Comment:
@clovisdasilvaneto Do we still need this haqrd-coded mask even using createAutoCorrectedDatePipe?

Reply:
@brunobasto Yes, I've tried to remove the mask property but the plugin need this property because the vanilhaTextMask makes a date parser based in the mask property